src/components/ui/ImageGallery/ImageGallerySelector.tsx

```diff
@@ -0,0 +1,130 @@
+import type { HTMLAttributes, PropsWithChildren } from 'react'
+import React, { useMemo } from 'react'
+import { useSlider, IconButton } from '@faststore/ui'
+
+import { ForwardArrowIcon, BackwardArrowIcon } from './Icons'
+
+interface Props extends HTMLAttributes<HTMLDivElement> {
+  itemsPerPage: number
+}
+
+const createTransformValues = (totalItems: number) => {
+  if (!totalItems) {
+    return {}
+  }
+
+  const transformMap: Record<number, number> = {}
+  const slideWidth = 100 / totalItems
+
+  for (let idx = 0; idx < totalItems; ++idx) {
+    const transformValue = -(slideWidth * idx)
+
+    transformMap[idx] = transformValue
+  }
+
+  return transformMap
+}
+
+function ImageGallerySelector({
+  itemsPerPage,
+  children,
+  ...otherProps
+}: PropsWithChildren<Props>) {
+  const elements = React.Children.toArray(children)
+  const elementCount = elements.length
+
+  const transformValues = useMemo(
+    () => createTransformValues(elementCount),
+    [elementCount]
+  )
+
+  const { handlers, slide, sliderState, sliderDispatch } = useSlider({
+    totalItems: elementCount,
+    itemsPerPage,
+    infiniteMode: false,
+  })
+
+  const slidingTransition = `transform 400ms`
+
+  if (!elements || !elementCount) {
+    return null
+  }
+
+  return (
+    <section>
+      <IconButton
+        aria-label="slide image selector backward"
+        icon={<BackwardArrowIcon color="#323845" />}
+        onClick={() => {
+          if (sliderState.sliding) {
+            return
+          }
+
+          slide('previous', sliderDispatch)
+        }}
+      />
+      <IconButton
+        aria-label="slide image selector forward"
+        icon={<ForwardArrowIcon color="#323845" />}
+        onClick={() => {
+          if (sliderState.sliding) {
+            return
+          }
+
+          slide('next', sliderDispatch)
+        }}
+      />
+
+      <div {...otherProps} {...handlers}>
+        <div
+          data-carousel-track
+          style={{
+            display: 'flex',
```

Comment:
Nice job here!
I think you have a plan to move the static CSS (the ones that don't depend on js values) in the style PR afterward, right?

Reply:
Yes, I think so.

Reply:
I'm curious, why will it be done separately?

Reply:
The ImageGallerySelector will not be present in the alpha version of the styled base.store. Since we are focusing on different component integrations/styling now, the styling for this component will be on hold for a while.